File: LicenseKeyFormatting2.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    string licenseKeyFormatting(string S, int K) {
        int count = 0;
        for(char c : S) count += (c != '-');
        int groups = count / K;
        count = count % K == 0 ? K : count - groups * K;
        string result = "";
        for(char c : S) {
        	if(c == '-') continue;
        	if(c >= 'a' && c <= 'z') c -= 32;
        	if(count-- == 0) {
        		result.push_back('-');
        		count = K - 1;
			}
			result.push_back(c);
		}
		return result;
        
    }
};

int main() {
	
	string S;
	cin>>S;
	
	int K;
	cin>>K;
	
	Solution *solution = new Solution();
	cout<<solution->licenseKeyFormatting(S, K);
	
	return 0;
}

File: LicenseKeyFormatting2.py
class Solution:
    def licenseKeyFormatting(self, S: str, K: int) -> str:
        S = S.replace('-', '').upper()
        l = len(S)
        s1 = l % K if l % K else K
        result = S[: s1]
        while s1 < l:
            result += '-' + S[s1: s1 + K]
            s1 += K
        return result


if __name__ == '__main__':
    S = input()
    K = int(input())
    solution = Solution()
    print(solution.licenseKeyFormatting(S, K))
